PSReadLine/Prediction.Views.cs

```diff
@@ -520,36 +619,254 @@ internal override void RenderSuggestion(List<StringBuilder> consoleBufferLines,
+                // to start render for the additional information part.
+                if (_selectedIndex > -1)
+                {
+                    for (int i = 0; i < _sources.Count; i++)
+                    {
+                        if (_selectedIndex <= _sources[i].EndIndex)
+                        {
+                            selected = i;
+                            break;
+                        }
+                    }
+
+                    if (selected == 0)
+                    {
+                        startFrom = 0;
+                    }
+                    else if (selected == _sources.Count - 1)
+                    {
+                        startFrom = Math.Max(0, selected - (maxSourceCount - 1));
+                    }
+                    else
+                    {
+                        startFrom = maxSourceCount == 3 ? selected - 1 : selected;
+                    }
+                }
+
+                // Start the extra information about the sources -- add the opening arrow bracket.
+                AppendColor(buffer, dimmedColor, ref activeColor, out _).Append('<');
+                additionalPartLength++;
+
+                // Add the prefix, continue to use dimmed color.
+                if (startFrom > 0)
+                {
+                    buffer.Append(SuggestionEntry.Ellipsis).Append(' ');
+                    additionalPartLength += 2;
+                }
+
+                // Add the sources.
+                for (int i = 0; i < maxSourceCount; i++)
+                {
+                    int index = startFrom + i;
+                    if (index == _sources.Count)
+                    {
+                        break;
+                    }
+
+                    if (i > 0)
+                    {
+                        // Add the separator.
+                        buffer.Append(' ');
+                        additionalPartLength++;
+                    }
+
+                    int nextCharPos;
+                    SourceInfo info = _sources[index];
+                    if (selected == index)
+                    {
+                        AppendColor(buffer, highlightColor, ref activeColor, out nextCharPos)
+                            .Append(info.SourceName)
+                            .Append('(')
+                            .Append(_selectedIndex - info.PrevSourceEndIndex)
+                            .Append('/')
+                            .Append(info.ItemCount)
+                            .Append(')');
+                    }
+                    else
+                    {
+                        AppendColor(buffer, dimmedColor, ref activeColor, out nextCharPos)
+                            .Append(info.SourceName)
+                            .Append('(')
+                            .Append(info.ItemCount)
+                            .Append(')');
+                    }
+
+                    // Need to take into account multi-cell characters when calculating length.
```

Comment:
Have you tested with double byte chars and or emojis?

Reply:
Yes, I did test with double byte characters:

![image](https://user-images.githubusercontent.com/127450/219745198-e1b917c8-4a16-465e-a0fd-d4f911b311e2.png)

But emojis are not guaranteed to work, because PSReadLine today doesn't handle surrogate pairs well. There is a label used to track all the emoji related issues.